[docs] Add a guide on how to deploy on railway (#3381)
mui/mui-toolpad: docs/data/toolpad/studio/how-to-guides/render-deploy.md

1. With a Render account, you can create a new **Web Service**:

{{"component": "modules/components/DocsImage.tsx", "src": "/static/toolpad/docs/studio/deploy-render/render-start.png", "alt": "Render new web service ", "caption": "Creating a new Render Web Service", "indent": 1 }}

2. Depending on whether the visibility of the GitHub repository is set to private or public, you need to either connect your GitHub account to Render, or paste in the URL of the repository you created above:

{{"component": "modules/components/DocsImage.tsx", "src": "/static/toolpad/docs/studio/deploy-render/render-connect-github.png", "alt": "GitHub new repo ", "caption": "Connecting GitHub to Render", "indent": 1, "aspectRatio": 2 }}

3. If using a private repository, you need to search for and select the repository you intend to deploy once connected to GitHub:

{{"component": "modules/components/DocsImage.tsx", "src": "/static/toolpad/docs/studio/deploy-render/render-search.png", "alt": "GitHub repo on Render ", "caption": "Connecting your GitHub repository", "indent": 1, "aspectRatio": 1.5}}

> 4. With the repository selected, you can advance to deployment configuration. Render can guess that you are deploying a Node app. Set the name which you want to appear in the URL of your deployed app, like `<APP-NAME>.onrender.com`:

Reviewer: ```suggestion
4. With the repository selected, you can advance to deployment configuration. Render can guess that you are deploying a Node app. Set the name which should appear in the URL of your deployed app, like `<APP-NAME>.onrender.com`:
```

That's it! We're up and running in a few minutes.

> Make changes, push to GitHub, and your app automatically redeploys each time. Like Render, you can deploy to any hosting provider of your choice.

Reviewer: ```suggestion
Make changes, push to GitHub, and your app automatically redeploys each time. You may deploy to any other hosting provider of your choice as well.
```
